### PR TITLE
influxctl reference documentation

### PR DESCRIPTION
**Pull request by sanderson**
Adds reference documentation for the `influxctl` CLI. There are some outstanding TODOs regarding installation, but the bulk of the content should be there.

- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+## Flags {#command-flags}
+
+| Flag        | Description                                                |
+| :---------- | :--------------------------------------------------------- |
+| `--profile` | Specify a connection profile to use (default is `default`) |
+
+---
+
+## Download and install influxctl
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md:48]: fwiw we are only providing a binary package download, so the instructions should be nearly the same (e.g. download this package, extract it, use the binary).

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+{{< /tabs-wrapper >}}
+
+## Configure connection profiles
+
+To connect with your InfluxDB Cloud Dedicated cluster, `influxctl` needs the
+following credentials:
+
+- InfluxDB Cloud Dedicated account ID
+- InfluxDB Cloud Dedicated cluster ID
+- InfluxDB Cloud Dedicated cluster URL
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md:86]: I've removed cluster URL from the current README, but I won't know for sure till I get access to a cluster to try it all. My understanding right now is we only talk to granite directly and so we do not need to know the URL

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+```sh
+influxctl database list [--format=json]
+```
+
+## Flags {#command-flags}
+
+| Flag       | Description                       |
+| :--------- | :-------------------------------- |
+| `--format` | Output format (default is `json`) |
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md:25]:
```suggestion
| `--format` | Output format (default is `table`) |
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+weight: 301
+---
+
+The `influxctl database list` command lists all database in an InfluxDB Cloud
+Dedicated cluster.
+
+## Usage
+
+```sh
+influxctl database list [--format=json]
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md:18]:
```suggestion
influxctl database list [--format=json|table]
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/token/list.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+weight: 301
+---
+
+The `influxctl token list` command lists all database tokens in an InfluxDB Cloud
+Dedicated cluster.
+
+## Usage
+
+```sh
+influxctl token list [--format=json]
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/token/list.md:18]:
```suggestion
influxctl token list [--format=table|json]
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/token/list.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+## Flags {#command-flags}
+
+| Flag       | Description                       |
+| :--------- | :-------------------------------- |
+| `--format` | Output format (default is `json`) |
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/token/list.md:25]:
```suggestion
| `--format` | Output format (default is `table`) |
```

**Review by powersj — COMMENTED** [review]
Thank you for putting these together! Couple comments below

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  cluster_id = "YOUR_OTHER_CLUSTER_ID"
+  cluster_url = "YOUR_OTHER_CLUSTER_URL"
+```
+
+## Authentication
+
+The `influxctl` CLI uses [Auth0](https://auth0.com/) to authenticate access to
+your InfluxDB Cloud Dedicated cluster.
+When you issue an `influxctl` command, the CLI checks for an active
+Auth0 token. If none exists, you are directed to login to **Auth0** via a
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md:121]:
```suggestion
**Auth0** token. If none exists, you are directed to login to **Auth0** via a
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+weight: 301
+---
+
+The `influxctl database create` command creates a new database in an InfluxDB
+Cloud Dedicated cluster.
+
+## Usage
+
+```sh
+influxctl database create [--retention-period 0s] {name}
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md:18]: > influxctl database create [--retention-period 0s] {name}


`{name}` or `<DATABASE_NAME>`?

Just wanna be consistent.

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+description: >
+  The `influxctl database list` command lists all database in an InfluxDB Cloud
+  Dedicated cluster.
+menu:
+  influxdb_cloud_dedicated:
+    parent: influxctl database
+weight: 301
+---
+
+The `influxctl database list` command lists all database in an InfluxDB Cloud
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md:12]:
```suggestion
The `influxctl database list` command lists all databases in an InfluxDB Cloud
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
+---
+title: influxctl database list
+description: >
+  The `influxctl database list` command lists all database in an InfluxDB Cloud
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/database/list.md:4]:
```suggestion
  The `influxctl database list` command lists all databases in an InfluxDB Cloud
```

**Review by jstirnaman — CHANGES_REQUESTED** [review]
Minor fix and a question. You cranked this out speedy-quick!

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md (hunk 8 above)
Comment (powersj) [reply content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md:53]: > My thinking here is that this is the command signature, not an example.

correct - my formatting was meant as an example of the command signature. Where '[]' indicated optional arguments and '{}' require required arguments.

Looking at a few man pages and the existing [influxdb cli](https://docs.influxdata.com/influxdb/cloud/reference/cli/influx/config/) docs it does look like I should have used `<name>` or `<database_name>` to specify a required argument.

How about we go with `<database_name>` if you want to keep this as a signature?

Comment (sanderson) [reply content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md]: Switched all the arguments to use the `<ARG>` syntax.

**Review by jstirnaman — APPROVED** [review]

**Pull request merged**